components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/java/org/wso2/carbon/mediator/enrich/ui/EnrichMediatorUtil.java

```diff
@@ -53,4 +54,77 @@ public static String getSynapsePathString(HttpSession session, SynapsePath synap
         return targetValue;
     }
 
+    /**
+     * Set the target expression of the enrich mediator.
+     *
+     * @param session        browser session
+     * @param enrichMediator mediator bean instance
+     * @param expressionId   expression id from request
+     * @param targetType     target type from request
+     */
+    public static void setTargetExpression(HttpSession session, EnrichMediator enrichMediator, String expressionId,
```

Comment:
Can't we further refactor this method to avoid duplicate code?

Reply:
I tried to replace setTargetExpression() setSourceExpression() methods with a single method but ended up with an unreadable code. Even though the structure looks similar in both methods, having target specific and source-specific parameters (like TARGET_EXPRESSION_ID) makes the method to have multiple conditional checks for Source/Target.

Reply:
However we can further refactor the two menthod avoiding duplicate code. @GDLMadushanka please update the PR with suggested changes.

Reply:
+1 done